### ISSUE TITLE
getExistingDiscoveryResult in DiscoveryServiceRegistryImpl return wrong result

### ISSUE THREAD

**Issue by Mr-Eskildsen** [eclipse/smarthome#5143]
When calling getExistingDiscoveryResult it requires a ThingUID, the ThingUID is allthough never used in the filtering, and the method just returns the first item in the list.

**Comment by maggu2810** [comment]
Hm, the code contains like:

```Java
ret = inboxReference.stream().filter(withFlag(DiscoveryResultFlag.NEW).and(forThingUID(thingUID))).collect(Collectors.toList());
```

For me it look like there is a filter for the flag `DiscoveryResultFlag.NEW` and one for the Thing UID `thingUID`.

What's missing?

**Comment by Mr-Eskildsen** [comment]
Arggghhh, haven't checked carefully enough. Seems that my local code is older than 13. november 2017.
The problem was actually fixed then in #4533. Before that the filtering on thingUID was missing:
`and(forThingUID(thingUID))).`

Case closed! Sorry for not looking carefull enough before reporting :-)
